### ISSUE TITLE
AttributeError: 'FigureCanvasAgg' object has no attribute 'renderer' / ValueError: Axis limits cannot be NaN or Inf

### ISSUE THREAD

**Issue by gaitay** [Kismuz/btgym#19]
Process DrawCerebro-2:1:
```
Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/home/git/btgym/btgym/rendering/plotter.py", line 75, in run
    figfilename='_tmp_btgym_render.png',
  File "/home/btgym/lib/python3.5/site-packages/backtrader/cerebro.py", line 991, in plot
    start=start, end=end, use=use)
  File "/home/btgym/lib/python3.5/site-packages/backtrader/plot/plot.py", line 220, in plot
    self.plotdata(data, self.dplotsover[data])
  File "/home/btgym/lib/python3.5/site-packages/backtrader/plot/plot.py", line 634, in plotdata
    data, opens, highs, lows, closes, volumes, vollabel)
  File "/home/btgym/lib/python3.5/site-packages/backtrader/plot/plot.py", line 580, in plotvolume
    ax.set_ylim(0, volylim, auto=True)
  File "/home/btgym/lib/python3.5/site-packages/matplotlib/axes/_base.py", line 3226, in set_ylim
    top = self._validate_converted_limits(top, self.convert_yunits)
  File "/home/btgym/lib/python3.5/site-packages/matplotlib/axes/_base.py", line 2836, in _validate_converted_limits
    raise ValueError("Axis limits cannot be NaN or Inf")
ValueError: Axis limits cannot be NaN or Inf
```

 
 

let self.volume = False then, 
```
Process DrawCerebro-2:1:
Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/home/git/btgym/btgym/rendering/plotter.py", line 78, in run
    rgb_string = fig.canvas.tostring_rgb()
  File "/home/btgym/lib/python3.5/site-packages/matplotlib/backends/backend_agg.py", line 460, in tostring_rgb
    return self.renderer.tostring_rgb()
AttributeError: 'FigureCanvasAgg' object has no attribute 'renderer'

```

**Comment by Kismuz** [comment]
@gaitay ,
can you provide more details on settings your are running? Does it happens with one of the examples or your own code and data?

**Comment by tmorgan4** [comment]
I just spent some time tracking down solutions to both issues. 

**ValueError: Axis limits cannot be NaN or Inf**
This is an issue caused by taking max() of an array full of NaNs and appears to be caused by a recent update.  It is in the process of being fixed.  For the time being, I changed line 130 in backtrader/plot/scheme.py from **self.volume = True** to **self.volume = False** to disable volume plots.

**AttributeError: 'FigureCanvasAgg' object has no attribute 'renderer'**
This is an easy fix but took quite a bit of time to track down.  Open btgym/rendering/plotter.py and add: **fig.canvas.draw()** before line 77 where rgb_string is called.

I'm new to Github and not familiar with pull requests so could someone else make this change?

**Comment by Kismuz** [comment]
@tmorgan4, @gaitay,
thank you for tracking this down. Indeed BTgym was developed to use Matplotlib 2.0.2 and crashes when 2.1 version is installed.

Solution:
The easiest way until `ValueError` is not fixed is to use matplotlib v/2.0.2
How to do:
It is highly recommended to run BTGym in designated virtual environment.
If you reinstalling btgym from scratch - just do nothing as v.2.0.2 is now specified in package dependencies and should install correct version of Matplotlib.
Else:
downgrade your matplotlib installation (that's where virtual env is handy):

    `pip install matplotlib==2.0.2`

Remark to @tmorgan4: I'm a bit paranoid about using `fig.canvas.draw()` as it potentially can lead to memory leaks. I had very though time fighting that issue when coding it initially. Anyway there is btgym/tests/render_memory_leak_test.ipynb you can try and see if it runs Ok.

**Comment by gaitay** [comment]
@Kismuz @tmorgan4 
Thank you for all your hard work!
